### PR TITLE
AP_Scheduler: Add check execute task time

### PR DESCRIPTION
**Pull request by muramura**
I will add a task. I want to measure the maximum processing time of the task. I added that mechanism.

### CODE REVIEW

File: libraries/AP_Scheduler/AP_Scheduler.cpp [hunk 1]
```diff
@@ -56,6 +56,21 @@ const AP_Param::GroupInfo AP_Scheduler::var_info[] = {
...
 
+    // @Param: TASK_NO
+    // @DisplayName: task number
+    // @Description: Specify the task number to measure the maximum execution time.
+    // @Range: 0 99
+    // @User: Standard
+    AP_GROUPINFO("TASK_NO",  2, AP_Scheduler, _task_no, 0),
+
+    // @Param: EXEC_TIME
+    // @DisplayName: the maximum execution time
```
Comment (amilcarlucas) [inline libraries/AP_Scheduler/AP_Scheduler.cpp]: This one should be marked as read only

File: libraries/AP_Scheduler/AP_Scheduler.cpp [hunk 2]
```diff
@@ -188,6 +203,14 @@ void AP_Scheduler::run(uint32_t time_available)
                   (unsigned)time_taken,
                   (unsigned)_task_time_allowed);
         }
+
+        if (_execTime[i] < time_taken) {
+            _execTime[i] = time_taken;
```
Comment (peterbarker) [inline libraries/AP_Scheduler/AP_Scheduler.cpp]: This is a memory-overwrite bug waiting to happen.

**Comment by rmackay9** [comment]
@muramura, thanks for the contribution.  We had a chat (on mumble) just now and we totally agree that we need to have some better tools for developers to allow them to investigate the time tasks take... but unfortunately we don't think that reporting this to the developer using a parameter is the way to go.  We probably need some sort of logging and/or mavlink message.

I guess you already know about the SCHED_DEBUG parameter?

I'm going to go ahead and close this but thanks again!

File: libraries/AP_Scheduler/AP_Scheduler.h [hunk 3]
```diff
@@ -203,6 +208,9 @@ class AP_Scheduler
 
     // bitmask bit which indicates if we should log PERF message to dataflash
     uint32_t _log_performance_bit;
+
+    //
+    uint32_t _execTime[100];
```
Comment (peterbarker) [inline libraries/AP_Scheduler/AP_Scheduler.h]: That's a whole chunk of memory.

**Pull request closed without merging**